modules/aks_node_pool/variables.tf

```diff
@@ -1,28 +1,36 @@
 # REQUIRED variables (must be set by caller of the module)
 
 variable "node_pool_name" {
-  type = string
+  description = "(Required) The name of the Node Pool which should be created within the Kubernetes Cluster. Changing this forces a new resource to be created."
+  type        = string
 }
 
 variable "aks_cluster_id" {
-  type = string
+  description = "(Required) The ID of the Kubernetes Cluster where this Node Pool should exist. Changing this forces a new resource to be created."
+  type        = string
 }
 
 variable "zones" {
-  type    = list(string)
-  default = []
+  description = " (Optional) Specifies a list of Availability Zones in which this Kubernetes Cluster Node Pool should be located. Changing this forces a new Kubernetes Cluster Node Pool to be created."
+  type        = list(string)
+  default     = []
 }
 
 variable "vnet_subnet_id" {
-  default = null
+  description = " (Optional) The ID of the Subnet where this Node Pool should exist. Changing this forces a new resource to be created."
+  type        = string
+  default     = null
 }
 
 variable "machine_type" {
-  type = string
+  description = " (Required) The SKU which should be used for the Virtual Machines used in this Node Pool. Changing this forces a new resource to be created."
```

Comment:
just a formatting question, noticed a leading space for these 3 descriptions

Reply:
Updated the descriptions in the latest commit.